core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/InProcessTelemetryChannel.java

```diff
@@ -62,195 +52,26 @@
-  private static final String MAX_MAX_TELEMETRY_BUFFER_CAPACITY_NAME = "MaxTelemetryBufferCapacity";
-
-  public static final int DEFAULT_FLUSH_BUFFER_TIMEOUT_IN_SECONDS = 5;
-  private static final int MIN_FLUSH_BUFFER_TIMEOUT_IN_SECONDS = 1;
-  private static final int MAX_FLUSH_BUFFER_TIMEOUT_IN_SECONDS = 300;
-  private static final String FLUSH_BUFFER_TIMEOUT_IN_SECONDS_NAME = "FlushIntervalInSeconds";
-
-    private final static String DEVELOPER_MODE_SYSTEM_PROPRETY_NAME = "APPLICATION_INSIGHTS_DEVELOPER_MODE";
-
-    private final static String DEVELOPER_MODE_NAME = "DeveloperMode";
-    private final static String ENDPOINT_ADDRESS_NAME = "EndpointAddress";
-    private final static String MAX_TRANSMISSION_STORAGE_CAPACITY_NAME = "MaxTransmissionStorageFilesCapacityInMB";
-
-    private boolean developerMode = false;
-    private static TransmitterFactory s_transmitterFactory;
-
-    private boolean stopped = false;
-
-    private TelemetriesTransmitter telemetriesTransmitter;
-
-    private TelemetryBuffer telemetryBuffer;
-    private TelemetrySampler telemetrySampler;
-
-    private static AtomicLong itemsSent = new AtomicLong(0);
+public final class InProcessTelemetryChannel extends ATelemetryChannel<String> {
 
     public InProcessTelemetryChannel() {
-        boolean developerMode = false;
-        try {
-            String developerModeAsString = System.getProperty(DEVELOPER_MODE_SYSTEM_PROPRETY_NAME);
-            if (!LocalStringsUtils.isNullOrEmpty(developerModeAsString)) {
-                developerMode = Boolean.valueOf(developerModeAsString);
-            }
-        } catch (Exception e) {
-            developerMode = false;
-            InternalLogger.INSTANCE.trace("%s generated exception in parsing, stack trace is %s", DEVELOPER_MODE_SYSTEM_PROPRETY_NAME, ExceptionUtils.getStackTrace(e));
-        }
-		initialize(null, null, developerMode, createDefaultMaxTelemetryBufferCapacityEnforcer(null),
-				createDefaultSendIntervalInSecondsEnforcer(null), true);
+        super();
     }
 
-  /**
-   * Ctor
-   *
-   * @param endpointAddress Must be empty string or a valid uri, else an exception will be thrown
-   * @param developerMode True will behave in a 'non-production' mode to ease the debugging
-   * @param maxTelemetryBufferCapacity Max number of Telemetries we keep in the buffer, when reached
-   *     we will send the buffer Note, value should be between TRANSMIT_BUFFER_MIN_TIMEOUT_IN_MILLIS
-   *     and TRANSMIT_BUFFER_MAX_TIMEOUT_IN_MILLIS inclusive
-   * @param sendIntervalInMillis The maximum number of milliseconds to wait before we send the
-   *     buffer Note, value should be between MIN_MAX_TELEMETRY_BUFFER_CAPACITY and
-   *     MAX_MAX_TELEMETRY_BUFFER_CAPACITY inclusive
-   */
-  public InProcessTelemetryChannel(
-      String endpointAddress,
-      boolean developerMode,
-      int maxTelemetryBufferCapacity,
-      int sendIntervalInMillis) {
-
-    // Builder pattern implementation
-    this(
```

Comment:
I don't follow.

Reply:
```
public InProcessTelemetryChannel(
      String endpointAddress,
      boolean developerMode,
      int maxTelemetryBufferCapacity,
      int sendIntervalInMillis)
```
This was one of the public constructors which was available. For SprinBoot starter I wanted another one:

```
public InProcessTelemetryChannel(
      String endpointAddress,
      String maxTransmissionStorageCapacity,
      boolean developerMode,
      int maxTelemetryBufferCapacity,
      int sendIntervalInMillis,
      boolean throttling,
      int maxInstantRetries)
```

Since we cannot deprecate a public constructor itself, therefore I used this kind of pattern to keep backportability. 

With your current change I don't see the second constructor anymore.

Reply:
it's on line 65.